src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,58 @@
 package core.basesyntax.service;
 
+import core.basesyntax.custom.exception.UserValidationException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int USER_MIN_AGE = 18;
+    private static final int USER_PASSWORD_MIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        String msg = checkForNullValues(user);
+        if (!msg.equals("OK")) {
+            throw new UserValidationException(msg);
+        } else if (storageDao.get(user.getLogin()) != null) {
+            throw new UserValidationException("Login is already registered");
+        } else if (!isLoginValid(user)) {
+            throw new UserValidationException(
+                    "Can't be empty or contain whitespace");
+        } else if (!isAgeValid(user)) {
+            throw new UserValidationException(
+                    "Must meet the minimum age requirement");
+        } else if (!isPasswordValid(user)) {
+            throw new UserValidationException(
+                    "Must be at least 6 characters without whitespace");
+        }
```

Comment:
It is better to check and throw exceptions in the appropriate method, it will be more readable
```suggestion
        checkForNullValues(user);
        loginValid(user);
        userExists(user);
        ageValid(user);
        passwordValid(user);
```

Reply:
Understood, thanks!